request_handlers.cc

```diff
@@ -15,10 +15,16 @@ char* request_handlers::echoRequestHandler(boost::shared_ptr<tcp::socket> socket
                                            size_t& response_buffer_size) {
   // Read in request
   char request_buffer[MAX_REQ_SIZE];
-  // TODO: Handle socket-read error
   boost::system::error_code ec;
   std::size_t bytes_read = socket->read_some(boost::asio::buffer(request_buffer), ec);
-  std::cout << "~~~~~~~~~~Request~~~~~~~~~~\n" << request_buffer << std::endl;
+  switch (ec.value()) {
```

Comment:
Nice use of a switch-statement! This gives a very elegant handling of the possible error codes.